eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php

```diff
@@ -836,6 +836,23 @@ public function testIndexingSpecialFullTextCases($text, $searchForText)
         $this->fail('Failed to find required Content in search results');
     }
 
+    /**
+     * Check if FullText indexing works for special cases of text.
+     *
+     * @param string $text Content Item field value text (to be indexed)
+     * @param string $searchForText text based on which Content Item should be found
+     * @param array $ignoreForSetupFactories list of SetupFactories to be ignored
+     * @dataProvider getLegacySpecialFullTextCases
+     */
+    public function testLegacyIndexingSpecialFullTextCases($text, $searchForText)
+    {
+        if (!$this->isLegacySearchEngineSetup()) {
```

Comment:
why?

Reply:
Because only Legacy Search uses the transformation Processor for indexing, which causes a problem and for some characters, as in the test, the resulting word is different than for SOLR or ElasticSearch

Reply:
> Because only Legacy Search uses the transformation Processor for indexing, which causes a problem and for some characters, as in the test, the resulting word is different than for SOLR or ElasticSearch

Unfortunately then this solution is not acceptable. The test in the form given was there to ensure such difference does not exist. Every engine needs to work uniformly for this use case.

Reply:
After verification, I do not see any reason why the transformations were performed during the indexing for legacy search. From what I checked, removing this does not cause any problems, and now consistency is maintained for search engines between legacy, SOLR, and Elastic - I tested it manually and it is possible to search correctly

Reply:
@Steveb-p @alongosz 
As for the case sensivity, the transformationProcessor with ruleGroup: lowercase has been added. If we want to have consistent results for SOLR ELASTIC and Legacy search, unfortunately, this transformation needs to be removed if we want consistent results. I have done tests on MySQL and also tests in PR for PostgreSQL, MySQL, and SQLite are green. Unfortunately, I do not see any other solution that would be able to have the same results for all supported search engines, the current solution that exists in the product already showed errors with a few special characters, which excludes them as correct